lib/modules/manager/pip-compile/extract.ts

```diff
@@ -200,6 +200,35 @@ export async function extractAllPackageFiles(
     depsBetweenFiles,
     packageFiles,
   );
+  for (const packageFile of result) {
+    for (const reqFile of packageFile.managerData?.requirementsFiles ?? []) {
+      if (fileMatches.includes(reqFile)) {
+        const sourceFile = result.find((packageFile) =>
+          packageFile.lockFiles?.includes(reqFile),
+        );
+        if (!sourceFile) {
+          logger.warn(
+            `pip-compile: ${packageFile.packageFile} references ${reqFile} which does not appear to be a requirements file managed by pip-compile`,
+          );
+          continue;
+        }
+        sourceFile.lockFiles = [
+          ...sourceFile.lockFiles!,
+          ...packageFile.lockFiles!,
+        ];
+      } else if (packageFiles.has(reqFile)) {
+        const sourceFile = packageFiles.get(reqFile)!;
+        sourceFile.lockFiles = [
+          ...sourceFile.lockFiles!,
+          ...packageFile.lockFiles!,
+        ];
+      } else {
+        logger.warn(
+          'pip-compile: references to requirements files not managed by pip-compile are not currently fully supported',
```

Comment:
```suggestion
          'pip-compile: references to requirements must be managed by pip-compile for Renovate to update the requirements files properly',
```

How about something like this?

Reply:
I wonder if it should just be the same warning as the case above.  It's a slightly different situation (`pip-compile->fileMatch` matched the file but it wan't able to read it in the first case vs `pip-compile->fileMatch` didn't match it at all in the second case), but maybe it's not necessary to distinguish between the two.

Reply:
Refactored so that both paths log the same warning.